### DISCUSSION TITLE
公式ドキュメントの不足を SwiftOnTap で補う。

### DISCUSSION THREAD

**Discussion by YusukeHosonuma** [YusukeHosonuma/Effective-SwiftUI#10] [Ideas]
## Overview

Apple の公式ドキュメントはサンプルコードやスクリーンショットが不足していることが多く、最初の大まかなコードの書き方や動作イメージを想像することが難しいケースが多い。

[SwiftOnTap](https://swiftontap.com/) は有志によって作成されたドキュメントであるものの、サンプルコードや動作イメージ（動画付きのものも多数）が非常に充実しており、利用者にとってわかりやすく優しい API ドキュメントに仕上がっている。

![image](https://user-images.githubusercontent.com/2990285/161880432-cbfe5fff-892a-4707-81a8-045e29aeca85.png)

## Discussion

最新の正確な API 仕様については、引き続き Apple の公式ドキュメントを参照すること。

SwiftOnTap はあくまで有志による作成なので、最新の API 仕様には追従できていないケースもある。また、すべてのコントロールについて十分なサンプルが網羅されてはいない。

コントリビューションも歓迎されているので、不足を感じたら PR を投げることで、より質の高いドキュメントに近づけることができるはず。

なお、以前に比べれば Apple の公式ドキュメントもサンプルコードやスクショが増えてきており、絶対的に不足しているというレベルからは少しずつ脱しつつある。

## Reference

- [SwiftOnTap](https://swiftontap.com/) 

## Note

[以前にバズったツイート](https://twitter.com/tobi462/status/1502150257479921667?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1502150257479921667%7Ctwgr%5E%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fzenn.dev%2F)
<img width="567" alt="image" src="https://user-images.githubusercontent.com/2990285/161880973-51ca14ec-5b4f-4c34-8aa6-9b1ca2edf0d2.png">

**Comment by tony1224** [comment]
@YusukeHosonuma 
今日そういえばと思って使おうと思ったのですが今見れなくなってるっぽいです
https://swiftontap.com/

公式ツイート
https://x.com/SwiftOnTap/status/1600661340028162048
